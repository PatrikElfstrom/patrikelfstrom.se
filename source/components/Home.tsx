import styled from '@emotion/styled';
import { ReactElement } from 'react';

const Wrapper = styled.div({
  position: 'absolute',
  top: '75%',
  left: '50%',
  transform: 'translate(-50%, -50%)',
  '& ::selection': {
    backgroundColor: '#092b3f',
    color: '#fff',
  },
});

const Header = styled.header({
  margin: '1rem',
});

const Heading = styled.h1({
  color: '#fff',
  margin: 0,
  fontWeight: 800,
});

const Main = styled.main({
  margin: '1rem',
});

const Link = styled.a({
  color: '#cacaca',
  textDecoration: 'none',
  transition: 'color 200ms ease-in-out',
  '&:hover': {
    color: '#fff',
  },
});

const Footer = styled.footer({
  margin: '1rem',
});

const Small = styled.small({
  color: '#949494',
});

export const Home = (): ReactElement => (
  <Wrapper>
    <Header>
      <Heading>Patrik ElfstrÃ¶m</Heading>
    </Header>
    <Main id="site">
      <Link href="mailto:me@patrikelfstrom.se">me@patrikelfstrom.se</Link>
    </Main>
    <Footer>
      <Small>Web Developer</Small>
    </Footer>
  </Wrapper>
);
