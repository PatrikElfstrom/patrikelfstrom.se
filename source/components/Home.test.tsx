import TestRenderer from 'react-test-renderer';
import { Home } from './Home';

describe('<Home />', () => {
  it('should match snapshot', () => {
    const tree = TestRenderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
