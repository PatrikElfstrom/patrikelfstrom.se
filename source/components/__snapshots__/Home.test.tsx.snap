// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match snapshot 1`] = `
<div
  className="css-1wvxrjq-Home--Wrapper e1k529477"
>
  <header
    className="css-2rdymk-Home--Header e1k529476"
  >
    <h1
      className="css-pwnmkz-Home--Heading e1k529475"
    >
      Patrik Elfstr√∂m
    </h1>
  </header>
  <main
    className="css-glx5v1-Home--Main e1k529474"
    id="site"
  >
    <a
      className="css-19y7a3k-Home--Link e1k529470"
      href="mailto:me@patrikelfstrom.se"
    >
      me@patrikelfstrom.se
    </a>
  </main>
  <footer
    className="css-3n3n6s-Home--Footer e1k529473"
  >
    <small
      className="css-92sczg-Home--Small e1k529472"
    >
      Web Developer
    </small>
  </footer>
  <aside
    className="css-1ttjkfr-Home--Aside e1k529471"
  >
    <a
      className="css-19y7a3k-Home--Link e1k529470"
      href="https://github.com/PatrikElfstrom"
      target="_blank"
    >
      <svg
        fill="none"
        height="43"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M22.48.25a21.72 21.72 0 00-6.87 42.33c1.09.2 1.49-.47 1.49-1.04l-.03-3.7c-6.04 1.31-7.32-2.91-7.32-2.91-.99-2.5-2.41-3.18-2.41-3.18-1.97-1.34.15-1.32.15-1.32 2.18.16 3.33 2.24 3.33 2.24 1.93 3.32 5.08 2.36 6.32 1.8.2-1.4.75-2.35 1.38-2.9-4.83-.55-9.9-2.4-9.9-10.73a8.4 8.4 0 012.24-5.83 7.8 7.8 0 01.21-5.75s1.82-.58 5.97 2.23a20.74 20.74 0 0110.88 0c4.14-2.81 5.96-2.23 5.96-2.23 1.19 3 .44 5.2.22 5.75a8.38 8.38 0 012.23 5.83c0 8.34-5.07 10.18-9.91 10.72.78.67 1.47 2 1.47 4.02 0 2.9-.03 5.24-.03 5.96 0 .58.4 1.25 1.5 1.04A21.73 21.73 0 0022.48.25z"
          fill="#fff"
          fillRule="evenodd"
        />
      </svg>
    </a>
  </aside>
</div>
`;
